File: server-e/app/Services/HealthMonitor.php
<?php

namespace App\Services;

use Illuminate\Support\Facades\Log;

class HealthMonitor
{
    public function __construct(
        private readonly DatabaseHealthChecker $databaseHealthChecker,
        private readonly HealthReporter $healthReporter,
    ) {
    }

    public function checkDatabase(): bool
    {
        if ($this->databaseHealthChecker->check()) {

            $this->healthReporter->healthy(
                'mysql',
                'DATABASE_AVAILABLE'
            );

            return true;
        }

        Log::error('Server E MySQL health check failed.');

        $this->healthReporter->down(
            'mysql',
            'DATABASE_UNAVAILABLE',
            'Unable to connect to MySQL.'
        );

        return false;
    }
}

File: server-d/api/services/health_monitor.py
import logging

from .database_health_checker import DatabaseHealthChecker
from .health_reporter import HealthReporter


logger = logging.getLogger(__name__)


class HealthMonitor:

    def __init__(
        self,
        database_health_checker=None,
        health_reporter=None,
    ):
        self.database_health_checker = (
            database_health_checker
            or DatabaseHealthChecker()
        )

        self.health_reporter = (
            health_reporter
            or HealthReporter()
        )

    def check_database(self):
        healthy = self.database_health_checker.check()

        if healthy:
            self.health_reporter.healthy(
                "mysql",
                "DATABASE_AVAILABLE",
            )

            return True

        logger.error(
            "Server D MySQL health check failed."
        )

        self.health_reporter.down(
            "mysql",
            "DATABASE_UNAVAILABLE",
            "Unable to connect to MySQL.",
        )

        return False